light-clients/cf-guest-cw/src/contract.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2022 ComposableFi.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use cosmwasm_std::{
+	to_json_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdError, StdResult, Uint64,
+};
+
+use crate::{context, context::log, ibc, msg, state};
+
+type Result<T = (), E = crate::error::Error> = core::result::Result<T, E>;
+
+#[cfg_attr(not(feature = "library"), cosmwasm_std::entry_point)]
+fn instantiate(
+	deps: DepsMut,
+	env: Env,
+	_info: MessageInfo,
+	msg: msg::InstantiateMsg,
+) -> Result<Response> {
+	let mut ctx = context::new(deps, env);
+	log!(ctx, "instantiate: {msg:?}");
+
+	ctx.client_states_mut().set(&msg.client_state);
+
+	let height = ibc::Height::new(0, msg.client_state.latest_height.into()).unwrap();
+	let metadata = ctx.metadata;
+	ctx.consensus_states_mut().set(height, &msg.consensus_state, metadata);
+
+	Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), cosmwasm_std::entry_point)]
+fn sudo(deps: DepsMut, env: Env, msg: msg::SudoMsg) -> Result<Response> {
```

Comment:
why are these messages processed in sudo?

Reply:
I don't recall the other light clients needing it

Reply:
I copied this from Tendermint: https://github.com/ComposableFi/composable-ibc/blob/master/light-clients/ics07-tendermint-cw/src/contract.rs#L171